src/components/Post/Post.jsx

```diff
@@ -1,66 +1,107 @@
-import React from "react";
+import React, { useState } from "react";
 import { useLocation, useNavigate } from "react-router-dom";
-import { AiOutlineHeart, AiOutlineComment, FiBookmark } from "assets/icons";
+import {
+  AiOutlineHeart,
+  AiOutlineComment,
+  FiBookmark,
+  AiOutlineEdit,
+  AiOutlineDelete,
+} from "assets/icons";
 import { IconButton } from "components";
+import { useDispatch, useSelector } from "react-redux";
+import { EditPostModal } from "components";
+import { deletePost } from "redux/reducers/postsSlice";
 
-const Post = () => {
+const Post = ({ ...params }) => {
+  const { image, caption, name, username, profile_pic, likes, comments, id } =
+    params;
```

Comment:
Suggestion: instead of getting the rest parameters from the props object and then destructruing from params object, you can directly destructure from props object